WIP: Improve look of documentation

TheZoq2 opened the pull request:
The documentation felt a bit too cluttered for me, so I changed it a bit. This is what it looks like rendered:

![image](https://user-images.githubusercontent.com/5312813/93707553-0a6ae100-fb30-11ea-94e8-180ca02b1b66.png)

I'm not a huge fan of the colors on the doc headers either, but that might be more of a personal preference. In any case, i'll try to come up with an alternataive

TheZoq2 committed:
Improve look of documentation

TheZoq2 commented:
![image](https://user-images.githubusercontent.com/5312813/93707756-96c9d380-fb31-11ea-9616-4544e6ca2307.png)
A proposal for reducing the amount of color. To me, this draws the eyes towards the actual text rather than the header bars.

TheZoq2 committed:
Slightly reduce the use of color

TheZoq2 committed:
Re-order documentation to put 'What is RIBS' first

TheZoq2 committed:
Add responsive sidebar

TheZoq2 commented:
I should probably focus more on the docs content, but this is a lot more fun :)

- Limited maximum width of content to avoid super long lines
- TOC to a floating sidebar if there is enough space
- Re-ordered documentation to put "What is RIBS" on top

![Peek 2020-09-20 12-00](https://user-images.githubusercontent.com/5312813/93708735-e790fa80-fb38-11ea-8e71-6e4751e70879.gif)

FredTheDino commented:
Nice! Looks a lot easier to parse now! Nice PR.

TheZoq2 committed:
Remove argument list from TOC

TheZoq2 commented:
One final thing, removed the argument list from the TOC to make that easier to parse :)

FredTheDino commented on README.md:
```diff
@@ -23,7 +23,7 @@ pip install -r requirements.txt
 ## Documentation
 
-The documentation for RIBS can be found at [tbd]
+The documentation for RIBS can be found at https://lithekod.github.io/snake-ribs/
```
Shouldn't this be a link?

TheZoq2 replied:
Yes, I assumed links were detected automagically when I wrote that, i'll take care of it

TheZoq2 replied:
Oh wait, this is markdown so it should be automatically made a link I think

TheZoq2 commented:
Hang on, something broke when I did something and now text is super small, going to look into it

FredTheDino reviewed (approved):
Great PR!

Would be nice with:
  - Some padding on the bottom of the ToC
  - A slightly larger monospace font.
  - Some way to differentiate code blocks from the other code, a different background or a border or something.

TheZoq2 committed:
Fix code tags not being escaped properly

TheZoq2 commented:
Looks like html got confused by the `<code>` tags being present in the ids and names of elements. Eventually it had enough and just filled the whole document with them :) Should be fixed now

TheZoq2 commented:
Never mind, still not fixed :scream:

TheZoq2 committed:
Fix code tag escaping 2 electric bogaloo

TheZoq2 commented:
Ok, fixed now (I think)

Making my own todo list:

- [x] Some padding on the bottom of the ToC
- [x] A slightly larger monospace font.
- [x] Some way to differentiate code blocks from the other code, a different background or a border or something.
- [x] Nicer looking links in the TOC
- [ ] Maybe highlight default arguments and keyword arguments in the titles
- [x] Darker code block background
- [ ] Fix ugly spacing after description blocks

TheZoq2 commented:
![image](https://user-images.githubusercontent.com/5312813/93709417-460ca780-fb3e-11ea-84c9-2e2ad1a4507a.png)

Moved the gray color to the code examples only. I think it's an improvement but I'm not sure

TheZoq2 committed:
Improve code sample style

FredTheDino commented:
I like it, makes them easier to parse! :+1:

sornas commented:
I agree with having grey only for code but I think it can be a bit darker.

TheZoq2 committed:
Improve sidebar style

TheZoq2 committed:
Add a bit more spacing inside descriptions

TheZoq2 commented:
![image](https://user-images.githubusercontent.com/5312813/93709635-11015480-fb40-11ea-987d-3faa480d7636.png)

Updated table of contents to get rid of ugly visited link color, improve font size and add a hover effect.

I also added a bit more spacing between headers and content. Next i'll look into fixing the ugly spacing after some descriptions

![image](https://user-images.githubusercontent.com/5312813/93709653-4c038800-fb40-11ea-8cf7-bc63b175cd9e.png)

TheZoq2 committed:
Some small doc fixups

TheZoq2 committed:
Slightly reduce font size of TOC header

TheZoq2 commented:
![image](https://user-images.githubusercontent.com/5312813/93710040-9d614680-fb43-11ea-923d-d28af4b6a577.png)
Made some final adjustments. There might be one or two things left that I *could* do, but i'm happy with this for now

EMattfolk commented:
I was thinking that the arguments should go right under function definitions, so as to get a quick overview of the function. I could do this if you guys think it's a good idea.

TheZoq2 commented:
I think that makes sense

TheZoq2 commented:
Rebased on master now, so this should be ready to merge

sornas commented:
It looks like not everything got the larger monospace font.

![image](https://user-images.githubusercontent.com/4735435/93710348-427d1e80-fb46-11ea-8565-305fd05a5c60.png)

FredTheDino reviewed (approved):
LGTM, if you want to fix the final piece of monospace front, that'd be great, unless I do it before you...

FredTheDino committed:
Fix monospace font for non-codeblock block
